src/scripts/postupgrade

```diff
@@ -1,3 +1,12 @@
-#!/bin/sh
+#!/bin/bash
+
+# If we are upgrading from DSM6->DSM7 we need to migrate state.
+# DSM6 builds are 1.X.Y-000Z, DSM7 builds are 1.X.Y-200Z
+if [[ "${SYNOPKG_OLD_PKGVER}" =~ "^1\.[0-9]*[1-9]\.[0-9]*[1-9]-0[0-9]{2}[1-9]$" ]]; then
```

Comment:
where's the Z in this regexp? I see a `$` but no `Z$`.

Reply:
i meant Z as in the last digit. like `1.12.3-2003` and `1.12.3-0003`

Reply:
Oh! Maybe use N and then say (where N is 1-9)

LGTM otherwise